### PR TITLE
Add UMI clustering

### LINKED ISSUE #16: Add UMIs
### Description of feature

Add the subworkflow to handle samples with UMIs

### PR DESCRIPTION
**Pull request by mirpedrol**
Add UMI clustering functionality.
Close #16 

## PR checklist

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the pipeline conventions in the [contribution docs](https://github.com/nf-core/crisprseq/tree/master/.github/CONTRIBUTING.md)- [ ] If necessary, also make a PR on the nf-core/crisprseq _branch_ on the [nf-core/test-datasets](https://github.com/nf-core/test-datasets) repository.
- [ ] Make sure your code lints (`nf-core lint`).
- [ ] Ensure the test suite passes (`nextflow run . -profile test,docker --outdir <OUTDIR>`).
- [ ] Usage Documentation in `docs/usage.md` is updated.
- [ ] Output Documentation in `docs/output.md` is updated.
- [ ] `CHANGELOG.md` is updated.
- [ ] `README.md` is updated (including new tool citations and authors/contributors).

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -7,6 +7,8 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Added
 
+- Add UMI clustering to crisprseq-edition ([#24](https://github.com/nf-core/crisprseq/pull/24))
```
Comment (mirpedrol) [inline CHANGELOG.md:10]:
```suggestion
- Add UMI clustering to crisprseq-targeted ([#24](https://github.com/nf-core/crisprseq/pull/24))
```

File: workflows/crisprseq_targeted.nf [hunk 2]
```diff
@@ -437,9 +678,9 @@ workflow CRISPRSEQ_TARGETED {
     //
     // MODULE: Parse cigar to find edits
     //
-    CIGAR_PARSER (
-        ch_to_parse_cigar
-    )
+    //CIGAR_PARSER (
+    //    ch_to_parse_cigar
+    //)
```
Comment (mirpedrol) [inline workflows/crisprseq_targeted.nf:683]:
```suggestion
    CIGAR_PARSER (
        ch_to_parse_cigar
    )
```

File: workflows/crisprseq_targeted.nf [hunk 3]
```diff
@@ -269,38 +324,224 @@ workflow CRISPRSEQ_TARGETED {
     }
 
 
-    /*
-    The UMI clustering step is posponed until the next release, the steps to be implemented are listed below:
-
     //
     // MODULE: Extract UMI sequences
     //
     EXTRACT_UMIS (
```
Comment (mirpedrol) [inline workflows/crisprseq_targeted.nf:330]: Add parameter to skip UMI clustering steps

File: nextflow_schema.json [hunk 4]
```diff
@@ -348,5 +428,11 @@
         {
             "$ref": "#/definitions/generic_options"
         }
-    ]
+    ],
+    "properties": {
+        "schema_ignore_params": {
```
Comment (mirpedrol) [inline nextflow_schema.json:433]: Move to the right definition

File: docs/usage_targeted.md
Comment (mirpedrol) [inline docs/usage_targeted.md:1]: Add description of pipeline parameters `--overrepresented` and the new parameter for umi clustering

**Review by LaurenceKuhl — APPROVED** [review]

**Pull request merged**